apps/backend/src/app/api/latest/emails/dev-server/chat/route.tsx

```diff
@@ -0,0 +1,196 @@
+import { createSmartRouteHandler } from "@/route-handlers/smart-route-handler";
+import { openai } from "@ai-sdk/openai";
+import { adaptSchema, yupArray, yupMixed, yupNumber, yupObject, yupString, yupUnion } from "@stackframe/stack-shared/dist/schema-fields";
+import { getEnvVariable } from "@stackframe/stack-shared/dist/utils/env";
+import { StatusError } from "@stackframe/stack-shared/dist/utils/errors";
+import { convertToCoreMessages, generateText, tool } from "ai";
+import { FreestyleSandboxes } from "freestyle-sandboxes";
+import { prismaClient } from "@/prisma-client";
+import { InferType } from "yup";
+import { z } from "zod";
+
+const textContentSchema = yupObject({
+  type: yupString().oneOf(["text"]).defined(),
+  text: yupString().defined(),
+});
+
+const toolCallContentSchema = yupObject({
+  type: yupString().oneOf(["tool-call"]).defined(),
+  toolName: yupString().defined(),
+  toolCallId: yupString().defined(),
+  args: yupMixed().defined(),
+  argsText: yupString().defined(),
+  result: yupMixed().defined(),
+});
+
+const contentSchema = yupArray(yupUnion(textContentSchema, toolCallContentSchema)).defined();
+
+
+export const POST = createSmartRouteHandler({
+  metadata: {
+    summary: "Chat with the dev server",
+    description: "Chat with the dev server to get help with email theme development",
+    tags: ["Emails"],
+  },
+  request: yupObject({
+    auth: yupObject({
+      type: yupString().oneOf(["admin"]).defined(),
+      tenancy: adaptSchema,
+    }),
+    body: yupObject({
+      repo_id: yupString().defined(),
+      messages: yupArray(yupObject({
+        role: yupString().oneOf(["user", "assistant"]).defined(),
+        content: yupString().defined(),
+      })).defined(),
```

Comment:
doing the minLength here will provide a better error message
```suggestion
      messages: yupArray(yupObject({
        role: yupString().oneOf(["user", "assistant"]).defined(),
        content: yupString().defined(),
      })).defined().minLength(1),
```